Assets/PatchKit Patcher/Scripts/AppUpdater/Status/DownloadSpeedCalculator.cs

```diff
@@ -64,14 +69,14 @@ public double BytesPerSecond
             get
             {
                 long bytes = _samples.Sum(s => s.Bytes);
-                long duration = _samples.Sum(s => s.Duration);
+                double duration = _samples.Sum(s => s.Duration.TotalSeconds);
 
-                if (bytes == 0)
+                if (bytes > 0 && duration > 0.0)
                 {
-                    return 0.0;
+                    return bytes / duration;
```

Comment:
Change it to `Math.Max(bytes / duration, 0)`. We don't want to see negative numbers here.

Reply:
There won't be negative numbers since `if (bytes > 0 && duration > 0.0)` is protecting from such situation.